src/modules/keyboardmanager/dll/KeyboardEventHandlers.cpp

```diff
@@ -32,6 +32,12 @@ namespace KeyboardEventHandlers
                     target = VK_LWIN;
                 }
 
+                // If Ctrl/Alt/Shift is being remapped to Caps Lock, then reset the modifier key state to fix issues in certain IME keyboards where the IME shortcut gets invoked since it detects that the modifier and Caps Lock is pressed even though it is suppressed by the hook - More information at the GitHub issue https://github.com/microsoft/PowerToys/issues/3397
```

Comment:
I understand the issue when Caps Lock to mapped to one of the modifier keys (say ctrl) but not the other way round. Why are we sending a `ctrl` key up msg in both the cases?

Reply:
If Ctrl is mapped to Caps Lock, at the moment when we do SendInput(CAPSLOCK), the IME will see both Ctrl and Caps Lock being pressed down, so before we send Caps Lock, we need to unset Ctrl (in the suppressed way so it doesn't affect anything else).